chapter3.adoc

```diff
@@ -62,16 +41,51 @@ The field `relaxedpriv` in abstractcs (at 0x16) allows for relaxed permission ch
+
+[NOTE]
+In scenarios where a Debug Module lacks System Bus Access, memory access by the debugger can be achieved through the use of abstract commands. These commands provide a secure means of accessing memory.
 
 [NOTE]
 Trusted entities like RoT should configure IOPMP or equivalent protection before granting debug access to machine mode. Similarly, machine mode should apply the protection before enabling submachine mode debug. 
 
 === Security Fault Error Reporting
 
-A dedicated error code, security fault error (cmderr 6), is included in cmderr of abstractcs (at 0x16) to signal any breaches of security enforcement by debug operations. Additionally, the bus security fault error (sberror 6) is introduced in sberror of sbcs (at 0x38) to denote errors related to system bus access.
+A dedicated error code, security fault error (cmderr 6), is included in cmderr of abstractcs (at 0x16) to signal any breaches of security enforcement by abstract commands. Additionally, the bus security fault error (sberror 6) is introduced in sberror of sbcs (at 0x38) to denote errors related to system bus access. The error raised by other debug operations which could be applied to multiple harts such as resethaltreq, reset and keepalive can be determined by the fields allsecfault/anysecfault in dmstatus. 
+
+The cmderr and sberror are defined as W1C (writing 1 to every bit clears the field) in RISC-V Debug Specification. Furthermore, any error indicated by allsecfault/anysecfault remains until updated through a successful resethaltreq, reset or keepalive operation. 
+
+=== Update of Debug Module Status (dmstatus, at 0x11)
+
+[caption="Register {counter:rimage}: ", reftext="Register {rimage}"]
+[title="The `allsecured` and `anysecured` fields in dmstatus"]
+[id=dmstatus]
+[wavedrom, ,svg]
+....
+{reg: [
+  {bits:   20, name: 'defined in Debug Module'},
+  {bits:   1, name: 'allsecured'},
+  {bits:   1, name: 'anysecured'},
+  {bits:   3, name: 'defined in Debug Module'},
+  {bits:   1, name: 'allsecfault'},
+  {bits:   1, name: 'anysecfault'},
+  {bits:   5, name: '0'},
+], config:{lanes: 3, hspace:1024}}
+....
+
+[[regdmstatus]]
+.The updates in dmstatus 
+[options="header"]
+|================================================================================================================================================
+| Field       | Description                                                                                                      | Access | Reset
+| allsecured  | The field is 1 when all currently selected harts implement Zedsec extension                                      | R      | -    
```

Comment:
For consistency with RISC-V Debug Spec, can you please update reset column with Preset instead of -?

Reply:
The value of all*/any* fields are produced dynamically according to selected harts. I don't think it has a platform-specific reset value. I think the "Preset" is for the deterministic fields.

Reply:
Your argument makes sense to me. Let me resolve this one.